pkg/provisioningv2/rke2/planner/planner_test.go

```diff
@@ -0,0 +1,120 @@
+package planner
+
+import (
+	"strings"
+	"testing"
+
+	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
+	"github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1/plan"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha4"
+)
+
+func TestPlanner_addInstruction(t *testing.T) {
+	type args struct {
+		version         string
+		expectedVersion string
+		os              string
+		command         string
+		scriptName      string
+		envs            []string
+	}
+
+	tests := []struct {
+		name string
+		args args
+	}{
+		{
+			name: "Checking Linux Instructions",
+			args: args{
+				version:         "v1.21.5+rke2r2",
+				expectedVersion: "v1.21.5-rke2r2",
+				os:              "linux",
+				command:         "sh",
+				scriptName:      "run.sh",
+				envs:            []string{"INSTALL_RKE2_EXEC"},
+			},
+		},
+		{
+			name: "Checking Windows Instructions",
+			args: args{
+				version:         "v1.21.5+rke2r2",
```

Comment:
Should we instead be using an rke2 version that supports the system-agent in RKE2?

Reply:
This is just a test to ensure that versions are translated correctly, the string doesn't really matter.